### PR TITLE
Define github workflow

### PR DESCRIPTION
**Pull request by Swarathmica-infraspec**
In this pull request, we have created github workflow

### CODE REVIEW

**Comment by kevinbheda** [comment]
can you please give brief description on what all changes are being done in this pull request ? coz I see github workflow etc..

File: .github/workflows/payoutManagementSystem.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          go-version: '1.22.2'
+          cache-dependency-path: go.sum
+      - name: Display Go version
+        run: go version
+      - name: Install dependencies
+        run: go get .
+      - name: Run main test
+        run: go test -v
+      - name: Run feature test
+        run: go test -v ./payee/
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .github/workflows/payoutManagementSystem.yml]: Resolved in commit "Refactor: Modify yml file"

File: payee/payee.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	if name == "" {
+		return nil, ErrEmptyName
+	}
+	if code == "" {
+		return nil, ErrEmptyCode
+	}
+	if numberOfDigits(accNumber) != 10 && numberOfDigits(accNumber) != 16 {
+		return nil, ErrInvalidAccountNumber
+	}
+	if numberOfDigits(mobile) != 10 {
```
[the remark this answers is not in the document]
Comment (KirthiInfra) [reply payee/payee.go:39]: Resolved in commit message as "Add tests as per boundary value analysis"

File: payee/payee.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+func NewPayee(name string, code string, accNumber int, ifsc string, bankName string,
+	email string, mobile int, payeeCategory string) (*payee, error) {
+	if name == "" {
+		return nil, ErrEmptyName
+	}
+	if code == "" {
+		return nil, ErrEmptyCode
+	}
+	if numberOfDigits(accNumber) != 10 && numberOfDigits(accNumber) != 16 {
```
[the remark this answers is not in the document]
Comment (KirthiInfra) [reply payee/payee.go:36]: Resolved in commit message as "Add tests as per boundary value analysis"

File: .github/workflows/payoutManagementSystem.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches:
+      - main
+      - payee
+  pull_request:
+    branches:
+      - main
+
+jobs:
```
Comment (kevinbheda) [inline .github/workflows/payoutManagementSystem.yml:12]: The github action says its "building and pushing" but action commands say otherwise.
Please update github action name to reveal exactly what the action is doing
Screenshot 2025-08-27 at 4 51 26 PM

**Review by kevinbheda — CHANGES_REQUESTED** [review]

File: .github/workflows/payoutManagementSystem.yml (hunk 4 above)
Comment (KirthiInfra) [reply .github/workflows/payoutManagementSystem.yml:12]: Resolved!

**Comment by kevinbheda** [comment]
> is this project a REST API or command line what is it ? can you elaborate in brief in the README ?

@KirthiInfra @Swarathmica-infraspec  can you update this please

**Comment by Swarathmica-infraspec** [comment]
> > is this project a REST API or command line what is it ? can you elaborate in brief in the README ?
> 
> @KirthiInfra @Swarathmica-infraspec can you update this please

Done

**Comment by Swarathmica-infraspec** [comment]
> > can you please give brief description on what all changes are being done in this pull request ? coz I see github workflow etc..
> 
> @Swarathmica-infraspec @KirthiInfra can you update this please?

Done

File: payee/payee_test.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	{"abc", "123", 6780002345765432, "cbin05648", "cbi", "abc@gmail.com", 9123456789, "Employee", nil},
+	{"abc", "123", 6780002345765432, "cbin05648", "cbi", "abc@gmail.com", 912345678, "Employee", ErrInvalidMobileNumber},
+	{"abc", "123", 6780002345765432, "cbin05648", "cbi", "abc.com", 9123456782, "Employee", ErrInvalidEmail},
+	{"", "123", 6780002345765432, "cbin05648", "cbi", "abc@gmail.com", 9123456782, "Employee", ErrEmptyName},
+	{"abc", "", 6780002345765432, "cbin05648", "cbi", "abc@gmail.com", 9123456782, "Employee", ErrEmptyCode},
+}
+
+func TestValidatePayee(t *testing.T) {
+	for _, tt := range errTests {
+		_, err := NewPayee(tt.beneficiaryName, tt.beneficiaryCode, tt.accNo, tt.ifsc, tt.bankName, tt.email, tt.mobile, tt.payeeCategory)
```
[the remark this answers is not in the document]
Comment (kevinbheda) [reply payee/payee_test.go:33]: @KirthiInfra @Swarathmica-infraspec t.fatal will not run other tests in the array errtests, please ensure that each of the tests inside errTests are run independently
https://go.dev/wiki/TableDrivenTests

**Review by kevinbheda — CHANGES_REQUESTED** [review]

File: .github/workflows/payoutManagementSystem.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22.2'
+          cache-dependency-path: go.sum
+      - name: Display Go version
+        run: go version
+      - name: Install dependencies
+        run: go get .
+      - name: Run main test
```
[the remark this answers is not in the document]
Comment (kevinbheda) [reply .github/workflows/payoutManagementSystem.yml]: @KirthiInfra  @Swarathmica-infraspec

File: .github/workflows/payoutManagementSystem.yml (hunk 4 above)
Comment (kevinbheda) [reply .github/workflows/payoutManagementSystem.yml:5]: github build action is not revealing intention

File: README.md [hunk 7]
```diff
@@ -1 +1,37 @@
...
+  - README.md <br>
+
+NOTE: Only email ids with .com are supported.
+
+# Run Tests
+
+Test can be run by executing the below command in the terminal
+  go test -v ./...
+
+NOTE: this project is still under development and hence does not have CLI or HTTP API.
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply README.md]: Resolved

File: .github/workflows/payoutManagementSystem.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .github/workflows/payoutManagementSystem.yml]: Done

**Comment by Swarathmica-infraspec** [comment]
> > is this project a REST API or command line what is it ? can you elaborate in brief in the README ?
> 
> @KirthiInfra @Swarathmica-infraspec this is still not addressed

done

File: payee/payee.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply payee/payee.go:37]: resolved

File: payee/payee.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply payee/payee.go:40]: resolved

File: payee/payee_test.go [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+	email           string
+	mobile          int
+	payeeCategory   string
+}{
+	{"abc", "123", 6780003045, "CBIN056489", "cbi", "abc@gmail.com", 9123456789, "Employee"},
+	{"abc", "123", 6780002345765432, "CBIN056489", "cbi", "abc@gmail.com", 9123456789, "Employee"},
+}
+
+func TestPayeeCanBeCreatedWithValidValues(t *testing.T) {
+	for _, tt := range validPayeeFieldsTests {
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply payee/payee_test.go:61]: done

File: payee/payee.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
+package payoutmanagementsystem
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply payee/payee.go:1]: resolved

File: .devcontainer/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,13 @@
+FROM mcr.microsoft.com/devcontainers/go:1-1.23-bookworm
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .devcontainer/Dockerfile:1]: This Dockerfile is only for the devcontainer. It’s not used in any GitHub workflow.

File: payee/payee.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (kevinbheda) [reply payee/payee.go:1]: what was the resolution? can you explain briefly ?

File: go.mod [hunk 11]
```diff
@@ -0,0 +1,3 @@
+module payoutmanagementsystem
+
+go 1.22.2
```
Comment (kevinbheda) [inline go.mod:3]: when was this go version released and when did we start this project ? 
what is the recommended version strategy when starting a new project ?

File: .github/workflows/payoutManagementSystem.yml [hunk 12]
```diff
@@ -0,0 +1,44 @@
+name: Payout Management System
+
+on: [push, pull_request]
+
+jobs:
+  format:
+    name: Check format
+    runs-on: ubuntu-22.04
```
Comment (kevinbheda) [inline .github/workflows/payoutManagementSystem.yml:8]: what is the recommended version of ubuntu suggested by github?
what strategy of version selection is recommeded ?

**Review by kevinbheda — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -1 +1,36 @@
...
+Test can be run by executing the below command in the terminal
+  go test -v ./...
+
+NOTE: this project is still under development and hence does not have HTTP API now.
+
+# CI
+
+The workflow is triggered on every push and pull request.
+It runs the following checks automatically:
+- Linting with `golangci-lint`
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply README.md:36]: resolved

File: .github/workflows/payoutManagementSystem.yml [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.22.2'
+          cache-dependency-path: go.sum
+      - run: test -z "$(gofmt -l .)"
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .github/workflows/payoutManagementSystem.yml:15]: gofmt is the tool for formatting go code. It has multiple options such as -l, -w, -d, etc. go fmt  internally runs gofmt -w -l .

gofmt is recommended in CI as we are checking if files are formatted and not formatting them.

File: .github/workflows/payoutManagementSystem.yml [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22.2'
+          cache-dependency-path: go.sum
+      - name: Display Go version
+        run: go version
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .github/workflows/payoutManagementSystem.yml]: It is no longer required. Removed.

File: .github/workflows/payoutManagementSystem.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (KirthiInfra) [reply .github/workflows/payoutManagementSystem.yml:15]: it is using the github hosted runner of image ubuntu, in our case, it is taken from the /bin.

File: .devcontainer/docker-compose.yml [hunk 16]
```diff
@@ -0,0 +1,38 @@
+version: '3.8'
+
+volumes:
+  postgres-data:
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .devcontainer/docker-compose.yml]: Removed

File: .github/workflows/payoutManagementSystem.yml (hunk 4 above)
Comment (kevinbheda) [reply .github/workflows/payoutManagementSystem.yml:5]: what was the resolution , can you please explain in brief

File: .github/workflows/payoutManagementSystem.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (kevinbheda) [reply .github/workflows/payoutManagementSystem.yml:22]: how to verify that "go fmt" runs internally "gofmt -w -l" ? can you share any reference ?

**Review by kevinbheda — CHANGES_REQUESTED** [review]

File: .github/workflows/payoutManagementSystem.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (kevinbheda) [reply .github/workflows/payoutManagementSystem.yml:22]: @KirthiInfra thats not correct info, I suppose in google chat you shared newer reference, can you please update it here ? thanks

**Review by kevinbheda — CHANGES_REQUESTED** [review]

File: .github/workflows/payoutManagementSystem.yml [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.25.0'
+          cache-dependency-path: go.sum
+      - name: Install dependencies
+        run: go mod tidy
```
Comment (kevinbheda) [inline .github/workflows/payoutManagementSystem.yml:52]: is this recommend to run on CI ?

**Review by kevinbheda — CHANGES_REQUESTED** [review]

File: payee/payee_test.go [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+	beneficiaryCode string
+	accNo           int
+	ifsc            string
+	bankName        string
+	email           string
+	mobile          int
+	payeeCategory   string
+	expectedErr     error
+}{
+	{"Validate Account Number Length", "invalid account number: 9 digits is used", "abc", "123", 678000234, "CBIN056489", "cbi", "abc@gmail.com", 9123456789, "Employee", ErrInvalidAccountNumber},
```
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply payee/payee_test.go]: Changed test names such that they reveal intention

File: .github/workflows/payoutManagementSystem.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (Swarathmica-infraspec) [reply .github/workflows/payoutManagementSystem.yml:22]: Tried man go fmt
image

[the remark this answers is not in the document]
Comment (KirthiInfra) [reply .github/workflows/payoutManagementSystem.yml:22]: https://linuxhandbook.com/shell-builtin-commands/

in this, we can find builtin commands of which one is test. basically it runs in shell of using binary file (/usr/bin/bash) on github hosted runner (ubuntu).

image

File: go.mod (hunk 11 above)
Comment (Swarathmica-infraspec) [reply go.mod]: We changed to 1.25.0
